### PR TITLE
Adding sync support - including dav tests

### PR DESCRIPTION
**Pull request by DeepDiver1975**
WebDAV Collection sync is required for addressbook sync for federated sharing #19566 

@evert the sync tests of CalDAVTester are failing because there is no status 200 added to the mutli status response and CalDavTester is rendering this as invalid - https://github.com/DeepDiver1975/CalDAVTester/blob/master/verifiers/multistatusItems.py#L90-L107

The spec hold the status as well - https://tools.ietf.org/html/rfc6578#section-3.8

requires https://github.com/owncloud/3rdparty/pull/222

### CODE REVIEW

File: apps/dav/lib/server.php [hunk 1]
```diff
@@ -39,6 +39,7 @@ public function __construct(IRequest $request, $baseUri) {
 		$this->server->addPlugin(new \OCA\DAV\Connector\Sabre\ExceptionLoggerPlugin('webdav', $logger));
 		$this->server->addPlugin(new \OCA\DAV\Connector\Sabre\LockPlugin());
 		$this->server->addPlugin(new \OCA\DAV\Connector\Sabre\ListenerPlugin($dispatcher));
+		$this->server->addPlugin(new \Sabre\DAV\Sync\Plugin());
```
Comment (PVince81) [inline apps/dav/lib/server.php]: do we need to restrict the plugin to a specific URL/endpoint ?

Comment (DeepDiver1975) [inline apps/dav/lib/server.php]: no - the plugin will only react on collections of a certain type

**Comment by DeepDiver1975** [comment]
blocked by https://github.com/fruux/sabre-dav/pull/734

**Comment by DeepDiver1975** [comment]
3rdparty merged

**Comment by DeepDiver1975** [comment]
failing integration test is unrelated

**Comment by LukasReschke** [comment]
:+1: :rocket:

**Comment by rullzer** [comment]
:+1:

**Pull request merged**